DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_changing_default_pvid.py

```diff
@@ -0,0 +1,104 @@
+                                                       configure_bridge_setup, configure_vlan_setup)
+
+
+pytestmark = [pytest.mark.suite_functional_ipv4,
+              pytest.mark.asyncio,
+              pytest.mark.usefixtures("cleanup_bridges", "cleanup_tgen")]
+
+packet_vids = ['X', 4094, 2]  # VLAN tag number contained in the transmitted packet
+port_map = ({"port": 0, "settings": [{"vlan": 4094, "untagged": True, "pvid": True}]},
+            {"port": 1, "settings": [{"vlan": 4094, "untagged": True, "pvid": True}]},
+            {"port": 2, "settings": [{"vlan": 4094, "untagged": True, "pvid": True}]},
+            {"port": 3, "settings": [{"vlan": 4094, "untagged": True, "pvid": True}]})
+
+
+async def test_vlan_changing_default_pvid(testbed):
+    """
+    Test Name: Changing default VLAN pvid
+    Test Suite: suite_functional_vlan
+    Test Overview: Test broadcast packet forwarding with non-default pvid
+    Test Procedure:
+    1. Initiate test params.
+    2. Set links to vlans.
+    3. Map receiving and non receiving dut_ports.
+        Port 1 ->  tx_port
+        Ports 2, 3, 4 -> rx_ports
+    4. Setup packet stream(s) for the broadcast packet:
+    5. Send traffic to rx_ports
+    6. Verify traffic on rx ports
+    """
+
+    # 1. Initiate test params.
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+    non_default_pvid = packet_vids[-1]
+
+    await configure_bridge_setup(device, dut_ports, default_pvid=non_default_pvid)
+
+    # 2. Set links to vlans.
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    # 3.Map receiving and non receiving dut_ports.
+    dev_groups = tgen_utils_dev_groups_from_config([
+        {"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+        {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, },
+        {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24, },
+        {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[1]][0]["name"],
+                dev_groups[tg_ports[2]][0]["name"],
+                dev_groups[tg_ports[3]][0]["name"]]
+
+    # 4. Setup packet stream(s).
+    streams = {f"Traffic with VLAN ID: {vlan}": {
+        "type": "raw",
+        "protocol": "802.1Q",
+        "ip_source": tx_ports,
+        "ip_destination": rx_ports,
+        "src_mac": "02:00:00:00:00:01",
+        "dst_mac": "ff:ff:ff:ff:ff:ff",
+        "vlanID": vlan
```

Comment:
If we are testing `pvid` do we need to send tagged traffic?

Reply:
To check that there was no changes in tagged traffic as well... ?

Reply:
okay, you are sending multiple streams here